code.js

```diff
@@ -1,3 +1,21 @@
 function permutationSort(a) {
-    return 0;
+    let counter = 0;
+    let swapped;
+    
+    do {
+        swapped = false;
+        for (var i = 0; i < a.length -1; i++) {
```

Comment:
This looks like a bubble sort.

Reply:
No, please see the description of the problem.

Reply:
I see now.  A new question i have now is do you want to us to use a java script random call to randomly orient the list or is there a different way you are thinking to do it.

Reply:
No, you have to produce all permutations systematically.

Reply:
I think i am a little confused on how to do this, is there a clue you could give me to get on the right track or an example

Reply:
You need to generate all possible permutations of the input list and check which one is sorted.

Reply:
I understand that part, but the part im not fully grasping is how you would generate the possible permutations without using a method that just starts at the first element and compares it to every other element and then moves to the second element and repeating this process until it finds it, sort of like a version of bubble sort which you said doesnt work.

Reply:
Well that's the exercises -- come up with code that generates all permutations. Note that you cannot generate all permutations by simply iterating through the elements one by one.